src/server/pkg/cmdutil/cobra.go

```diff
@@ -292,6 +294,47 @@ func ParseHistory(history string) (int64, error) {
 	return int64(result), err
 }
 
+// ParseJobFilters parses filter strings into JobFilters
+// Currently each string must be the name of a job state
+func ParseJobFilters(filterStrs []string) (string, error) {
+	var conditions []string
+	for _, stateStr := range filterStrs {
+		var state pps.JobState
+		if value, ok := pps.JobState_value[stateStr]; ok {
+			state = pps.JobState(value)
+		} else {
+			switch strings.ToLower(stateStr) {
+			case "starting":
+				state = pps.JobState_JOB_STARTING
+			case "running":
+				state = pps.JobState_JOB_RUNNING
+			case "merging":
+				state = pps.JobState_JOB_MERGING
+			case "failure":
+				state = pps.JobState_JOB_FAILURE
+			case "success":
+				state = pps.JobState_JOB_SUCCESS
+			case "killed":
+				state = pps.JobState_JOB_KILLED
+			default:
+				return "", fmt.Errorf("states must be one of: starting, running, merging, failure, success, killed")
+			}
+		}
+		conditions = append(conditions, fmt.Sprintf(".state == \"%s\"", state))
+	}
+
+	out := fmt.Sprintf("if %s then . else null end", strings.Join(conditions, " or "))
+	q, err := gojq.Parse(out)
+	if err != nil {
+		return "", err
+	}
+	_, err = gojq.Compile(q)
+	if err != nil {
+		return "", err
+	}
+	return out, nil
+}
```

Comment:
You should write a test that uses this function to send requests against the server. We have some tests for pachctl but generally most of the test coverage comes from tests that directly leverage the api since pachctl is a pretty thin layer around it. I think it'd be fine to test just this function and send the request to the api rather than testing it through pachctl, which would probably be a little more annoying to do.

Reply:
That makes sense, I wrote a test along those lines, though I haven't tried running it locally yet.

Reply:
done, I have no strong opinions about file placement yet